### ISSUE TITLE
(docs) With which permission level is a user allowed to create a resource?

### ISSUE THREAD

**Issue by musicEnfanthen** [dasch-swiss/dsp-tools#132]
From the docs about permissions ([https://dasch-swiss.github.io/dsp-tools/dsp-tools-xmlupload/#describing-permissions-with-permissions-elements](https://dasch-swiss.github.io/dsp-tools/dsp-tools-xmlupload/#describing-permissions-with-permissions-elements)), it is unclear what permission level is needed for a user to create a resource.

**Comment by irinaschubert** [comment]
See this paragraph in the documentation:

Note: The permissions defined in the XML are applied to resources that are created. But only project or system administrators do have the permission to create resources via the XML upload.

**Comment by musicEnfanthen** [comment]
Yes, I saw that but it is still unclear to me. As far as I understand it, the paragraph in the documentation refers to the process of the XML upload. 

My question rather concerns the later usage of the permissions. If a user/user group is given permission right "V", they are only allowed to "view" a resource which should exclude the right to create a new resource, correct? If "M" is applied, they are allowed to "modify" an existing resource, but are they also allowed to create a new entry in the database (resource or property)? Or would they need "delete" permission to be able to also create a resource?

**Comment by irinaschubert** [comment]
Ah, I see. In order to create a resource over the app (admin.dasch.swiss) a user has to be project member or project admin. Creating resources with dsp-tools xmlupload functionality can only be done by a project admin. This has nothing to do with permissions defined in the XML file. These permissions apply only to the resources defined in that XML. Does this help?

**Comment by musicEnfanthen** [comment]
My apologies for not expressing the question clear enough, my fault. Since there are different levels in the process where permissions are involved, I try to separate concerns for now:

1. XML upload: project admin can create new resource CLASSES and resource class INSTANCES. That's clear from the docs.

2. admin.dasch.swiss: Project admin can create a resource (new CLASSES and INSTANCES, right?). That's also fine and totally reasonable.

3. admin.dasch.swiss: Project members can create new resources (CLASS or INSTANCE ?)? --> This is not clear from the docs. Where does this permission come from? Just from the status of being a project member? What if, for instance, a project member has the (honestly strange) restriction to have only `view` permissions on data of a project (for example, a group of proofreaders that are not allowed to change the data itself). So they would get the read-only permission from setting "V" on the resource instance (data) itself, but at the same time, they would be allowed to create new resource instances (i.e., create new data) because they have the status project member? 

This last use case is where my confusion comes from and what this issue is targeting.

EDIT: Re-thinking my question, I guess that the last use case is not relevant for the XML upload or any processes where dsp-tools is involved in. But it is relevant for the future usage of the uploaded data. So I guess my question finally condenses to: Are permissions only applied on the level of the data instances (with the mechanism described in the docs), or is there a possibility to restrict an entire user group to a certain permission stage in the future (such as proof-readers can only read, but neither modify nor create new resources)? If so, do these permission settings have to be defined anywhere in the ontology or elsewhere (probably not in the XML) before the upload?

Please let me know if the question is still unclear.

**Comment by irinaschubert** [comment]
No problem, that's what discussions are here for.

1. Correct
2. Correct: Project admin can create CLASSES and INSTANCES. 
3. Project member can create INSTANCES (not CLASSES) by default. A project member has this permissions just from being a project member. Yes, your assumption is right: a group of proofreaders would be able to view existing resources AND create new ones in that use case you describe.

Exactly, the permission to CREATE instances is not defined within the XML and has nothing to do with the permissions defined ON that instances. We distinguish two types of permissions: administrative permissions (create/edit/delete users, groups, projects, resource instances etc.) and object access permissions. Here is the documentation about it: https://docs.dasch.swiss/DSP-API/05-internals/design/api-admin/administration/#permissions

Yes, it is possible to change this behavior. You can define your own group whose members would be able to view (or edit or whatever) but not create resource instances. But at the moment, this can't be done over admin.dasch.swiss or dsp-tools. It needs to be done directly over the API. It would probably be best if you design the groups you want to have and then someone of DaSCH could create these special groups with the respective permissions for your project.

Administrative permissions can be changed easily but object access permissions can only be changed per resource instance. So, once you have a lot of resources and want to change the access permission for a group, you would have to change this on every single instance. Of course, you (or someone of DaSCH) could do that with a script (but still, it is quite some work someone has to do).
